src/main/scala/org/ergoplatform/nodeView/mempool/OrderedTxPool.scala

```diff
@@ -118,9 +116,9 @@ final case class OrderedTxPool(orderedTransactions: TreeMap[WeightedTxId, ErgoTr
     * @param weight
```

Comment:
Why not  create a graph of transactions where edge X -> Y iff Y spends some output from X. 
And then update the graph as new transactions are added.
Then the algorithm will be fully deterministic and more flexible for future improvements and extensions.

Reply:
Makes sense, but @knizhnik did not take this approach. Let's make low-priority issue on that?

Reply:
We actually have such graph now! The only question is how edges of this graph are represented. Right now there are no direct links between transaction. Instead of it we have list of input/output boxes IDs in transaction and map of box-id to transaction which produced it. 

Certainly storing direct links can eliminate extra map lookups and speed-up algorithm. The question is how critical it is...   When I setup new node and load about 100k transactions, only <10 of them are off-chain. 

So may be there some other use cases where there are much more off-chain transactions. In any case before optimizing this code, we should first prove that it is a bottleneck.

Also just having materialized graph doesn't help itself to order transactions by priority, because we should take in account not only weight of transaction itself, but also weight of all dependent transactions.

Reply:
I suggested to have graph explicitly not for optimization but for clarity. The dependencies are strict constraints which you cannot violate, the priorities are some additional weighting above the graph. When you prioritize and maybe shuffle around the pool you should strictly respect the graph. Once we have dependencies in the pool (which is a set of items) which we should respect, all algorithms over pool should respect and can utilize the graph. So why not have it explicitly. Again, not for the sake of performance, but for clarity. The performance will be a byproduct of this.

Reply:
I've made an issue about that https://github.com/ergoplatform/ergo/issues/1051